include/pybind11/cast.h

```diff
@@ -1870,7 +1870,12 @@ struct arg_v : arg {
 #if !defined(NDEBUG)
         , type(type_id<T>())
 #endif
-    { }
+    {
+      // Workaround! See https://github.com/pybind/pybind11/issues/2336
```

Comment:
Is this the correct issue number? I looked and it wasn't obvious.

The description of this PR currently references two different issues: #2682, #2422

Reply:
Yes. See https://github.com/pybind/pybind11/issues/2422#issuecomment-709642855

Reply:
Suggestion to link directly to https://github.com/pybind/pybind11/pull/2685#issuecomment-731286700 from here.

Reply:
Done.